api/capi/src/nntrainer.cpp

```diff
@@ -1286,7 +1286,10 @@ int ml_train_model_get_weight(ml_train_model_h model, const char *layer_name,
   for (unsigned int i = 0; i < dims.size(); ++i) {
     status = ml_tensors_data_set_tensor_data(
       *weight, i, w[i], dims[i].getDataLen() * sizeof(float));
-    return status;
+    if (status != ML_ERROR_NONE) {
+	    ml_tensors_data_destroy(weight);
```

Comment:
```
int ml_tensors_data_destroy (ml_tensors_data_h data)
```
and
```
ml_tensors_data_h *weight
```

Is this valid???

Reply:
Have you turned compiler-warning-error off?

Reply:
The parameters of ml_tensors_data_destroy' and 'ml_tensors_info_destroy' are being used in the wrong type, so we modified it.